src/main/java/picard/sam/markduplicates/util/AbstractMarkDuplicatesCommandLineProgram.java

```diff
@@ -311,5 +338,27 @@ private static void trackOpticalDuplicates(final List<? extends ReadEnds> list,
         if (opticalDuplicates > 0) {
             opticalDuplicatesByLibraryId.increment(list.get(0).getLibraryId(), opticalDuplicates);
         }
+        return opticalDuplicates;
     }
+
+    private static void trackDuplicateCounts(final int listSize,
+                                             final int optDupCnt,
+                                             final Histogram<Double> duplicatesCountHist,
+                                             final Histogram<Double> nonOpticalDuplicatesCountHist,
+                                             final Histogram<Double> opticalDuplicatesCountHist) {
+        duplicatesCountHist.increment((double)(listSize));
+        if ((listSize - optDupCnt) > 0) {
+            nonOpticalDuplicatesCountHist.increment((double) (listSize - optDupCnt));
+        }
+        if (optDupCnt > 0) {
+            opticalDuplicatesCountHist.increment((double) (optDupCnt + 1.0));
+        }
+    }
+
+    private static void addSingletonToCount(final Histogram<Double> duplicatesCountHist,
```

Comment:
As far as I can tell, this method is only called from one place, line 311 above. If that's true you can remove this copy and inline it above.